guides/micronaut-data-jdbc-repository/java/src/main/java/example/micronaut/GenreController.java

```diff
@@ -0,0 +1,84 @@
+package example.micronaut;
+
+import example.micronaut.domain.Genre;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.scheduling.TaskExecutors;
+import io.micronaut.scheduling.annotation.ExecuteOn;
+import io.micronaut.data.exceptions.DataAccessException;
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+
+@ExecuteOn(TaskExecutors.IO)  // <1>
+@Controller("/genres")  // <2>
+public class GenreController {
+
+    protected final GenreRepository genreRepository;
+
+    public GenreController(GenreRepository genreRepository) { // <3>
+        this.genreRepository = genreRepository;
+    }
+
+    @Get("/{id}") // <4>
+    public Genre show(Long id) {
+        return genreRepository
+                .findById(id)
+                .orElse(null); // <5>
+    }
+
+    @Put // <6>
+    public HttpResponse update(@Body @Valid GenreUpdateCommand command) { // <7>
+        int numberOfEntitiesUpdated = genreRepository.update(command.getId(), command.getName());
+
+        return HttpResponse
+                .noContent()
+                .header(HttpHeaders.LOCATION, location(command.getId()).getPath()); // <8>
+    }
+
+    @Get(value = "/list{?args*}") // <9>
+    public List<Genre> list(@Valid SortingAndOrderArguments args) {
```

Comment:
Should use `Pageable`

Reply:
I've changed to `Pageable` as a controller method argument. 

A couple of changes, I had to do:

- When binding Pageable as a method argument sort is bound as `sort=name,desc` in our examples we had something like `sort=name&order=desc'.

We had a test for an invalid order. 

```java
@Test
    public void supplyAnInvalidOrderTriggersValidationFailure() {
        HttpClientResponseException thrown = assertThrows(HttpClientResponseException.class, () -> {
            client.toBlocking().exchange(HttpRequest.GET("/genres/list?order=foo"));
        });

        assertNotNull(thrown.getResponse());
        assertEquals(HttpStatus.BAD_REQUEST, thrown.getStatus());
    }
```

Rewriting it as follows does not thrown any validation exception but binds to the default sorting which is `ASC`


```java
@Test
    public void supplyAnInvalidOrderTriggersValidationFailure() {
        HttpClientResponseException thrown = assertThrows(HttpClientResponseException.class, () -> {
            client.toBlocking().exchange(HttpRequest.GET("/genres/list?order=name,foo"));
        });

        assertNotNull(thrown.getResponse());
        assertEquals(HttpStatus.BAD_REQUEST, thrown.getStatus());
    }
```

I have removed the test.